### PR TITLE
Complete card 12

### PR DESCRIPTION
**Pull request by SimonRosellys**
Card 12 - GET /api/articles/:article_id completed.

### CODE REVIEW

**Comment by Liam310** [comment]
Hey Simon! Looks good, I'm going to list some general points here and some more specific points in the form of comments on the code (they will also show up here).

- Error nicely handled for the article_id which doesn't exist, but what about if I made a request to `/api/articles/notAnId`?
- At some point a nice refactor would be to have separate folders inside your `controllers` and `models` folders, i.e. one for all controllers related to articles, another one for all controllers related to topics and so on.

I'll let you know when I've finished leaving comments!

File: controllers/controllers.js [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
+      const specificArticle = {
+        author: article.name,
+        title: article.title,
+        article_id: Number(article_id),
+        body: article.body,
+        topic: article.topic,
+        created_at: article.created_at,
+        votes: article.votes,
+      };
+      res.status(200).send(specificArticle);
```
Comment (Liam310) [inline controllers/controllers.js]: A note on separation of concerns: the controllers are intended to handled requests and responses whilst the models are for interacting with the data. If the data needs to be at all formatted by the time it gets to the controller, that suggests that the model isn't quite doing its job properly (although in this case I suspect we shouldn't have to convert the `article_id` to a number here!)

File: models/models.js [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
+    .query(
+      `SELECT * FROM articles JOIN users ON articles.author = users.username`
+    )
+    .then((article) => {
+      if (article_id > article.rows.length) {
+        return Promise.reject({ status: 404, msg: "article does not exist" });
+      }
+      return article.rows[article_id - 1];
+    });
+};
```
Comment (Liam310) [inline models/models.js]: A couple of points to note here:
- We only want to `JOIN` tables together when we need to display/use the data from some other table. In this case, all we want do at the moment is get info about a particular article - is it necessary to join it onto the `users` table?
- Imagine if there were 1,000,000 articles in our database (for a website like reddit, for example). Currently this model would fetch all 1,000,000 of them just to send info about 1 of them back to the client. Perhaps there is a way to tell SQL to only give us the article we want?

**Comment by Liam310** [comment]
Okay, that's all my comments for this PR! Have a read and a think about how to tackle these matters 🙂

**Comment by cavalierswain** [comment]
Nice refactor! Looks good!

**Pull request merged**